### PR TITLE
Gradient Part 5c: Fix gradient rotation on text & math

### PR DESCRIPTION
**Pull request by Dherse**
Math mode uses a different transform which was not compatible with the old way of handling gradients on text. Now there is an added parameter on `TextItem` which keeps track of whether the text item is part of a math equation. I would have preferred if math and text shared the transform "system" but I can't figure out why they differ especially since this is specific to PDF. In any case, that should fix the issue while remaining compatible.

### CODE REVIEW

**Comment by Dherse** [comment]
Ok now it's truly fixed in a way that should also avoid any future bug that are PDF transform related.

**Comment by Dherse** [comment]
This actually fixes a bunch of potential bugs with gradients on text that nobody seems to have noticed yet. At least they're already fixed, sorry for introducing bugs, but transforms are hard *sigh*

File: crates/typst/src/geom/gradient.rs [hunk 1]
```diff
@@ -157,13 +158,23 @@ use crate::syntax::{Span, Spanned};
...
 #[ty(scope)]
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[derive(Clone, PartialEq, Eq, Hash)]
 pub enum Gradient {
     Linear(Arc<LinearGradient>),
     Radial(Arc<RadialGradient>),
     Conic(Arc<ConicGradient>),
 }
 
+impl fmt::Debug for Gradient {
```
Comment (laurmaedje) [inline crates/typst/src/geom/gradient.rs:168]: Why this change? Since we split up Debug and Repr specifically to not have to override Rust's Debug style with the Typst repr.

File: tests/typ/visualize/gradient-math.typ [hunk 2]
```diff
@@ -66,3 +65,27 @@ $ attach(
   Pi, t: alpha, b: beta,
   tl: 1, tr: 2+3, bl: 4+5, br: 6,
 ) $
+
+---
+
```
Comment (laurmaedje) [inline tests/typ/visualize/gradient-math.typ:70]: General style nit: I'd prefer not blank lines between "---" and the comment describing the test.

File: crates/typst/src/geom/gradient.rs (hunk 1 above)
Comment (Dherse) [reply crates/typst/src/geom/gradient.rs]: It was an artefact of me going frame-by-frame and transform-by-transform to figure out what was wrong, removed ;)

**Comment by laurmaedje** [comment]
Thank you, it's great than you managed to make it simpler instead of more complicated than the previous state. :)

**Comment by Dherse** [comment]
> Thank you, it's great than you managed to make it simpler instead of more complicated than the previous state. :)

@laurmaedje I am mostly glad I managed to fix it "at the source" and prevent other similar bugs from eventually creeping up!
